pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -133,6 +144,96 @@ var _ = Describe("MachineSet Reconciler", func() {
 			return ready, nil
 		}, timeout*3).Should(BeEquivalentTo(replicas))
 	})
+
+	It("Should set the Paused condition appropriately", func() {
+		instance := machineSetBuilder.
+			WithName("baz").
+			WithLabels(map[string]string{"baz": "bar"}).
+			Build()
+
+		By("Creating the MachineSet")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
+
+		By("Setting the AuthoritativeAPI to ClusterAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityClusterAPI
+		})).Should(Succeed())
+
+		By("Verifying that the AuthoritativeAPI is set to Cluster API")
+		Eventually(k.Object(instance), timeout).Should(HaveField("Status.AuthoritativeAPI", Equal(machinev1.MachineAuthorityClusterAPI)))
+
+		By("Verifying the paused condition is approproately set to true")
+		Eventually(k.Object(instance), timeout).Should(HaveField("Status.Conditions", ContainElement(SatisfyAll(
+			HaveField("Type", Equal(PausedCondition)),
+			HaveField("Status", Equal(corev1.ConditionTrue)),
+		))))
+
+		By("Transitioning the AuthoritativeAPI though 'Migrating' to MachineAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMigrating
+		})).Should(Succeed())
+
+		// The condition should remain true whilst transitioning through 'Migrating'
+		// Run this in a goroutine so we don't block
+
+		// Copy the instance before starting the goroutine, to avoid data races
+		instanceCopy := instance.DeepCopy()
+		go func() {
+			defer GinkgoRecover()
+			framework.RunCheckUntil(
+				ctx,
+				// Check that we consistently have the Paused condition true
+				func(_ context.Context, g framework.GomegaAssertions) bool {
+					By("Checking that the paused condition is consistently true whilst migrating to MachineAPI")
+
+					localInstance := instanceCopy.DeepCopy()
+					if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(localInstance), localInstance); err != nil {
+						return g.Expect(err).Should(WithTransform(isRetryableAPIError, BeTrue()), "expected temporary error while getting machine: %v", err)
+					}
+
+					return g.Expect(localInstance.Status.Conditions).Should(ContainElement(SatisfyAll(
+						HaveField("Type", Equal(PausedCondition)),
+						HaveField("Status", Equal(corev1.ConditionTrue)),
+					)))
+
+				},
+				// Condition / until function: until we observe the MachineAuthority being MAPI
+				func(_ context.Context, g framework.GomegaAssertions) bool {
+					By("Checking that the AuthoritativeAPI is not MachineAPI")
+
+					localInstance := instanceCopy.DeepCopy()
+					if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(localInstance), localInstance); err != nil {
+						return g.Expect(err).Should(WithTransform(isRetryableAPIError, BeTrue()), "expected temporary error while getting machine: %v", err)
+					}
+
+					return g.Expect(localInstance.Status.AuthoritativeAPI).ToNot(Equal(machinev1.MachineAuthorityMachineAPI))
```

Comment:
Are you sure you wanted `ToNot` here?

I'm a bit confused as we set the authoritativeAPI to not MAPI before we run this, so the condition will immediately pass no?

Reply:
Hmm - I think you may be right here. My bad 🤦  It should be a `To`. 

That way we will run the check ensuring that `Paused: True` until we observe  `Status.AuthoritativeAPI` *is* `MachineAuthorityMachineAPI`, at which point we expect it to be false. 


Come to think of it, I wonder if we want to set this `RunCheckUntil` off before we update the status to migrating..